src/Randomizer.App/Windows/TrackerWindow.xaml.cs

```diff
@@ -947,89 +951,34 @@ private async void LoadSavedStateMenuItem_Click(object sender, RoutedEventArgs e
             // If there is a valid rom, then load the state from the db
             if (GeneratedRom.IsValid(Rom))
             {
-                Tracker.Load(Rom);
+                await Task.Run(() => Tracker.Load(Rom));
+                
                 Tracker.StartTimer(true);
                 if (_dispatcherTimer.IsEnabled)
                 {
                     _dispatcherTimer.Start();
                 }
             }
-            // Otherwise save it to a file
             else
             {
-                var dialog = new OpenFileDialog
-                {
-                    Filter = "Tracker state files (*.json.gz)|*.json.gz|All files (*.*)|*.*"
-                };
-
-                while (dialog.ShowDialog(this) == true)
-                {
-                    try
-                    {
-                        using (var stream = new FileStream(dialog.FileName, FileMode.Open, FileAccess.Read, FileShare.Read))
-                        using (var gzip = new GZipStream(stream, CompressionMode.Decompress))
-                        {
-                            await Tracker.LoadAsync(gzip);
-                        }
-
-                        break;
-                    }
-                    catch (ArgumentException ex)
-                    {
-                        _logger.LogError(ex, "Failed to load Tracker state from '{fileName}'.", dialog.FileName);
-
-                        var result = MessageBox.Show(this, "Could not load Tracker using the selected saved state file. Please check you selected the right file and try again.", "SMZ3 Cas’ Randomizer", MessageBoxButton.OKCancel, MessageBoxImage.Warning);
-                        if (result == MessageBoxResult.Cancel)
-                            break;
-                    }
-                }
+                MessageBox.Show(this, "Could not save tracker state.", "SMZ3 Cas’ Randomizer", MessageBoxButton.OK, MessageBoxImage.Error);
```

Comment:
Please see if you can improve this error message a little too. I can imagine Betus or someone seeing this error and having no clue what to do.

Reply:
tbh I'm not sure what to do either. lol It's sort of a catch all that I don't think would ever happen, but I added it just as a safety net as if the rom were never actually loaded it would cause a good amount of issues loading the state. I can add a general message about posting the logs on Discord or Github like we do with unhandled exceptions.